### PR TITLE
Document anomalies under read-committed.

### PR DESCRIPTION
**Pull request by sherfert**
Slightly restructure locking section to accommodate these changes.

### CODE REVIEW

File: modules/ROOT/pages/database-internals/concurrent-data-access.adoc [hunk 1]
```diff
@@ -106,6 +100,61 @@ REMOVE n._LOCK_
...
+====
+
+The easiest way to work around this is to only read each property once, and keep it as long as needed in the query.
+
+=== Missing and double reads
+
+When scanning an xref:performance/index-configuration.adoc[index], entities may be observed multiple times or skipped entirely, even if they are present in the index.
+This is true even for indexes that back link:{neo4j-docs-base-uri}/cypher-manual/{page-version}/constraints/examples/#constraints-examples-node-uniqueness[uniqueness constraints].
+
+During the scan, if another concurrent query changes a property of an entity in the index, the entity might appear again if the property was changed to a position ahead of the scan.
```
Comment (renetapopova) [inline modules/ROOT/pages/database-internals/concurrent-data-access.adoc:130]:
```suggestion
During the scan, if another concurrent query changes an entity's property to a position ahead of the scan, the entity might appear again in the index.
```

File: modules/ROOT/pages/database-internals/concurrent-data-access.adoc [hunk 2]
```diff
@@ -106,6 +100,61 @@ REMOVE n._LOCK_
...
+Similarly, the entity might not appear at all if the property changed to a position that has already been scanned.
```
Comment (renetapopova) [inline modules/ROOT/pages/database-internals/concurrent-data-access.adoc:131]:
```suggestion
Similarly, the entity may not appear at all if the property is changed to a previously scanned position.
```

File: modules/ROOT/pages/database-internals/concurrent-data-access.adoc [hunk 3]
```diff
@@ -106,6 +100,61 @@ REMOVE n._LOCK_
...
+
+This anomaly can only occur with operators that scan an index, or parts of an index, for example link:{neo4j-docs-base-uri}/cypher-manual/{page-version}/planning-and-tuning/operators/operators-detail/#query-plan-node-index-scan[`NodeIndexScan`] or link:{neo4j-docs-base-uri}/cypher-manual/{page-version}/planning-and-tuning/operators/operators-detail/#query-plan-directed-relationship-index-seek-by-range[`DirectedRelationshipIndexSeekByRange`].
+
+.Missing and double read
+====
+In the following query one could expect to see every node `n` that has the property `prop` exactly once.
```
Comment (renetapopova) [inline modules/ROOT/pages/database-internals/concurrent-data-access.adoc:137]:
```suggestion
In the following query you can expect to see every node `n` that has the property `prop` exactly once.
```

File: modules/ROOT/pages/database-internals/concurrent-data-access.adoc [hunk 4]
```diff
@@ -106,6 +100,61 @@ REMOVE n._LOCK_
...
+However, concurrent updates that change the `prop` property while the index is being scanned, can lead to a node appearing multiple times or not at all in the result set.
```
Comment (renetapopova) [inline modules/ROOT/pages/database-internals/concurrent-data-access.adoc:138]:
```suggestion
However, concurrent updates that modify the `prop` property during index scanning may cause a node to appear multiple times or not at all in the result set.
```

**Review by renetapopova — APPROVED** [review]
Thanks, @sherfert! Looks really good. I added some editorial comments mainly related to style and simplicity.

**Comment by renetapopova** [comment]
@sherfert To which Neo4j version do these changes apply? We need to add a label to the PR.

**Review by Dunfluff — APPROVED** [review]
LGTM

File: modules/ROOT/pages/database-internals/concurrent-data-access.adoc (hunk 3 above)
Comment (sherfert) [reply modules/ROOT/pages/database-internals/concurrent-data-access.adoc:137]: I wanted to point out that "seeing every node `n` that has the property `prop` exactly once" could be considered a reasonable assumption. But, as it turns out, that assumption does not hold.

Using "you can expect" delivers the wrong message, doesn't it. I was also not entirely happy with "one could expect". Do you maybe have another suggestion?

**Comment by sherfert** [comment]
> @sherfert To which Neo4j version do these changes apply? We need to add a label to the PR.

This is documenting behavior that been this way since the dawn of time. So I would recommend merging it to 4.4 and whichever 5.X is easiest for you.

**Comment by sherfert** [comment]
@renetapopova 
Your suggestion seems to have disappeared - unless it is to delete the whole block:

Screenshot from 2024-05-13 09-33-54

**Comment by renetapopova** [comment]
> @renetapopova Your suggestion seems to have disappeared - unless it is to delete the whole block:
> 
> Screenshot from 2024-05-13 09-33-54

I am deleting just one empty line here.

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-1601.surge.sh

Updated pages:
- [Configuration settings](https://neo4j-docs-operations-1601.surge.sh/operations-manual/5/configuration/configuration-settings)
- [Database internals and transactional behavior](https://neo4j-docs-operations-1601.surge.sh/operations-manual/5/database-internals)
- [Transaction management](https://neo4j-docs-operations-1601.surge.sh/operations-manual/5/database-internals/transaction-management)

**Pull request merged**